### PR TITLE
feat(repocop): Evaluate all repositories

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change, and why?
We were previously evaluating one repository. This change evaluates all repositories, excluding those that are archived or outside of P&E. We're performing this filtering via SQL, which though isn't unit tested, does mean we hold less in memory, that is, it should be more performant.

## How has it been verified?
Ran locally. I've also [deployed to CODE](https://riffraff.gutools.co.uk/deployment/view/43f216e2-25f5-46c8-804a-7dd1e15cdb57), and successfully executed the lambda.

### CODE REVIEW

File: packages/repocop/src/index.ts [hunk 1]
```diff
@@ -12,26 +12,23 @@ export async function main() {
...
-		const ruleEvaluation = repositoryRuleEvaluation(repo, branches);
+	const data = await Promise.all(
+		repositories.map(async (repo) => {
+			const branches = await prisma.github_repository_branches.findMany();
+			return repositoryRuleEvaluation(repo, branches);
+		}),
+	);
 
-		console.log('The results are in...');
-		console.log(JSON.stringify(ruleEvaluation, null, 2));
```
Comment (akash1810) [inline packages/repocop/src/index.ts:25]: Logging this out for all repositories will be a bit too much, so removed.

**Comment by akash1810** [comment]
On CODE, this is failing with:

```log
START RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844 Version: $LATEST
2023-09-12T11:04:22.603Z	59795d59-a11e-4ec7-ba18-27bf7d8c0844	INFO	Generating RDS token
2023-09-12T11:04:27.933Z	59795d59-a11e-4ec7-ba18-27bf7d8c0844	INFO	Found 3438 repositories
RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844 Error: Runtime exited with error: signal: killed
Runtime.ExitError
END RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844
REPORT RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844	Duration: 17374.35 ms	Billed Duration: 17375 ms	Memory Size: 512 MB	Max Memory Used: 512 MB	Init Duration: 470.10 ms	
```

File: packages/cdk/lib/service-catalogue.ts [hunk 2]
```diff
@@ -500,6 +500,7 @@ export class ServiceCatalogue extends GuStack {
 			environment: { DATABASE_HOSTNAME: db.dbInstanceEndpointAddress },
 			vpc,
 			securityGroups: [applicationToPostgresSecurityGroup],
+			timeout: Duration.minutes(15),
```
Comment (akash1810) [inline packages/cdk/lib/service-catalogue.ts:503]: This branch on CODE was causing the lambda to fail with:

```log
START RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844 Version: $LATEST
2023-09-12T11:04:22.603Z	59795d59-a11e-4ec7-ba18-27bf7d8c0844	INFO	Generating RDS token
2023-09-12T11:04:27.933Z	59795d59-a11e-4ec7-ba18-27bf7d8c0844	INFO	Found 3438 repositories
RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844 Error: Runtime exited with error: signal: killed
Runtime.ExitError
END RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844
REPORT RequestId: 59795d59-a11e-4ec7-ba18-27bf7d8c0844	Duration: 17374.35 ms	Billed Duration: 17375 ms	Memory Size: 512 MB	Max Memory Used: 512 MB	Init Duration: 470.10 ms	
```

Update the timeout to 15 minutes out of laziness; 15 minutes is the maximum duration for AWS lambda. If we're still timing out, then we need to revise our approach, either by using smaller queries, or using a different runtime.

File: packages/repocop/src/index.ts [hunk 3]
```diff
@@ -1,7 +1,100 @@
...
+		},
+
+		// For archived repositories, we only need to know the repository name.
+		select: {
+			full_name: true,
+		},
+	});
+
+	console.log(
+		`Found ${repositories.length} archived repositories. Marking them all as passing.`,
```
[the remark this answers is not in the document]
Comment (akash1810) [reply packages/repocop/src/index.ts:36]: Yeah, good point!

**Review by tjsilver — APPROVED** [review]
The `where:NOT:OR` syntax in `getRepositories` is a little hard to parse but I guess that's how Prisma likes it
